### PR TITLE
174355181, 174355195 sign up with credit card

### PR DESCRIPTION
**Pull request by ShaidullinArtur**
Stories:
- https://www.pivotaltracker.com/story/show/174355181
- https://www.pivotaltracker.com/story/show/174355195

Note:
stripe public key: `pk_test_51H87YpHXoVM8VrBPcm5zJKttU4sEbq42g62e0kUQ746x8dnsHqbMxDzn0JaRAsj51Ku3fLF5tiBPYdEeNk4Xp4gc004A4dpZQw`

### CODE REVIEW

File: app/controllers/communitarian/payment_intents_controller.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      render_payment_intent_result(created_payment_intent)
+    end
+
+    private
+
+    def created_payment_intent
+      Communitarian::Stripe.new.created_payment_intent(payment_intent_params)
+    end
+
+    def render_payment_intent_result(result)
```
Comment (Vadim-Kurnatovskiy) [inline app/controllers/communitarian/payment_intents_controller.rb:18]: I think this is an unnecessary method, we use this code once.

**Comment by Vadim-Kurnatovskiy** [comment]
I think that "Next verification" inscription didn't need here
Снимок экрана от 2020-09-02 16-02-00

**Comment by Vadim-Kurnatovskiy** [comment]
Frame is too large
Снимок экрана от 2020-09-02 16-01-09

File: assets/javascripts/initializers/communitarian.js.es6 [hunk 2]
```diff
@@ -90,6 +93,24 @@ function initializeCommunitarian(api) {
   });
 
   api.modifyClass("controller:create-account", {
+    onShow() {
+      this.setProperties({
+        accountUsername: Date.now()
```
Comment (Vadim-Kurnatovskiy) [inline assets/javascripts/initializers/communitarian.js.es6]: hm, is it a good? it's a temporarily value for username validation skip?

**Comment by ShaidullinArtur** [comment]
@Vadim-Kurnatovskiy 

https://github.com/fs/communitarian-discourse-plugin/pull/33#issuecomment-685716809 - we have the same text in design. I simply copied it from there
Снимок экрана 2020-09-02 в 16 57 39

https://github.com/fs/communitarian-discourse-plugin/pull/33#issuecomment-685720077 - I don't have any issues with styles
Снимок экрана 2020-09-02 в 17 14 44

File: assets/javascripts/initializers/communitarian.js.es6 (hunk 2 above)
Comment (ShaidullinArtur) [reply assets/javascripts/initializers/communitarian.js.es6]: @Vadim-Kurnatovskiy yes, you're right. We updating username right after success sign up.

File: config/settings.yml [hunk 3]
```diff
@@ -32,6 +32,12 @@ plugins:
   landing_categories_length:
     default: 4
     client: true
+  sign_up_with_credit_card:
+    default: true
+    client: true
+  sign_up_with_stripe_identity:
```
Comment (DmitryBarskov) [inline config/settings.yml]: Maybe call them with _enabled at the end?

**Comment by IlgizGilmanov** [comment]
> @Vadim-Kurnatovskiy
> 
> [#33 (comment)](https://github.com/fs/communitarian-discourse-plugin/pull/33#issuecomment-685716809) - we have the same text in design. I simply copied it from there
> Снимок экрана 2020-09-02 в 16 57 39
> 
> [#33 (comment)](https://github.com/fs/communitarian-discourse-plugin/pull/33#issuecomment-685720077) - I don't have any issues with styles
> Снимок экрана 2020-09-02 в 17 14 44

This problem with modal styles occurs when the screen height is low. I'll fix it in the next PR.

File: assets/javascripts/initializers/communitarian.js.es6 [hunk 4]
```diff
@@ -90,6 +93,24 @@ function initializeCommunitarian(api) {
   });
 
   api.modifyClass("controller:create-account", {
```
Comment (DmitryBarskov) [inline assets/javascripts/initializers/communitarian.js.es6]: Maybe move this modifyClass invocation to another initializer to reduce file size?

**Review by IlgizGilmanov — APPROVED** [review]

File: assets/javascripts/discourse/controllers/payment-details.js.es6 [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+      if (result.error) {
+        self._showError(result.error.message);
+      } else {
+        self.set("paymentConfirmed", true);
+        self._createAccount()
+      }
+    });
+  },
+  _createAccount() {
+    var attrs = {
```
Comment (DmitryBarskov) [inline assets/javascripts/discourse/controllers/payment-details.js.es6]:
```suggestion
    let attrs = {
```

File: assets/javascripts/discourse/controllers/payment-details.js.es6 [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+      name: "",
+    });
+    this._createPaymentIntent();
+  },
+
+  actions: {
+    submitPaymentDetails(event) {
+      if (event) event.preventDefault();
+      if (this.submitDisabled) return;
+      this.set("loading", true)
```
Comment (DmitryBarskov) [inline assets/javascripts/discourse/controllers/payment-details.js.es6]:
```suggestion
      this.set("loading", true);
```

**Review by DmitryBarskov — APPROVED** [review]

**Pull request merged**